dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -641,35 +642,75 @@ protected void refreshTargetEntry(String path) {
 
     @Override
     protected void handlePublishUnpublish(String entryPath, String newName, boolean unpublishRequest) {
-        Workflow existingWorkflow;
+        Workflow existingWorkflow = null;
         boolean isPublished = false;
+
+        // Cannot be making an unpublish request with a specified new name
+        assert (!(unpublishRequest && newName != null));
+
         try {
             existingWorkflow = workflowsApi.getWorkflowByPath(entryPath, null, false);
             isPublished = existingWorkflow.isIsPublished();
         } catch (ApiException ex) {
-            exceptionMessage(ex, "Unable to publish/unpublish " + newName, Client.API_ERROR);
+            exceptionMessage(ex, "Unable to " + (unpublishRequest ? "unpublish " : "publish ") + entryPath, Client.API_ERROR);
+        }
+
+        if (existingWorkflow == null) {
+            errorMessage("Unable to locate " + entryPath, COMMAND_ERROR);
         }
+
         if (unpublishRequest) {
             if (isPublished) {
                 publish(false, entryPath);
             } else {
-                out("This workflow is already unpublished.");
+                out("The following workflow is already unpublished: " + entryPath);
             }
         } else {
             if (newName == null) {
                 if (isPublished) {
-                    out("This workflow is already published.");
+                    out("The following workflow is already published: " + entryPath);
                 } else {
                     publish(true, entryPath);
                 }
+            } else if (!workflowExists(entryPath + "/" + newName)) {
+                try {
+                    // path should be represented as repository organization and name (ex. dockstore/dockstore-ui2)
+                    final Workflow newWorkflow = workflowsApi.manualRegister(
+                            getGitRegistry(existingWorkflow.getGitUrl()),
+                            existingWorkflow.getOrganization() + "/" + existingWorkflow.getRepository(),
+                            existingWorkflow.getWorkflowPath(),
+                            newName,
+                            existingWorkflow.getDescriptorType().toString(),
+                            existingWorkflow.getDefaultTestParameterFilePath()
+                    );
+
+                    final String completeEntryPath = entryPath + "/" + newName;
+
+                    if (newWorkflow != null) {
```

Comment:
Won't newWorkflow always be set? If line 678 failed, won't it throw an exception?

Reply:
Ideally, yes. I added this because if, in some strange case, the `manualRegister()` API call had null as its return value, that would not throw an exception from the register function itself (We would hit a null pointer exception 2 lines later, so it wouldn't make it very far).

Either way, it is redundant, I will remove it from both tools and workflows.